pkg/services/ngalert/api/api.go

```diff
@@ -181,60 +192,117 @@ func (api *apiImpl) createAlertDefinitionEndpoint(c *models.ReqContext, cmd save
 
-	if err := api.store.getOrgAlertDefinitions(&query); err != nil {
+	if err := api.Store.GetOrgAlertDefinitions(&query); err != nil {
 		return response.Error(500, "Failed to list alert definitions", err)
 	}
 
 	return response.JSON(200, util.DynMap{"results": query.Result})
 }
 
-func (api *apiImpl) pauseScheduler() response.Response {
-	err := api.schedule.Pause()
+func (api *API) pauseScheduler() response.Response {
+	err := api.Schedule.Pause()
 	if err != nil {
 		return response.Error(500, "Failed to pause scheduler", err)
 	}
 	return response.JSON(200, util.DynMap{"message": "alert definition scheduler paused"})
 }
 
-func (api *apiImpl) unpauseScheduler() response.Response {
-	err := api.schedule.Unpause()
+func (api *API) unpauseScheduler() response.Response {
+	err := api.Schedule.Unpause()
 	if err != nil {
 		return response.Error(500, "Failed to unpause scheduler", err)
 	}
 	return response.JSON(200, util.DynMap{"message": "alert definition scheduler unpaused"})
 }
 
 // alertDefinitionPauseEndpoint handles POST /api/alert-definitions/pause.
-func (api *apiImpl) alertDefinitionPauseEndpoint(c *models.ReqContext, cmd updateAlertDefinitionPausedCommand) response.Response {
+func (api *API) alertDefinitionPauseEndpoint(c *models.ReqContext, cmd ngmodels.UpdateAlertDefinitionPausedCommand) response.Response {
 	cmd.OrgID = c.SignedInUser.OrgId
 	cmd.Paused = true
 
-	err := api.store.updateAlertDefinitionPaused(&cmd)
+	err := api.Store.UpdateAlertDefinitionPaused(&cmd)
 	if err != nil {
 		return response.Error(500, "Failed to pause alert definition", err)
 	}
 	return response.JSON(200, util.DynMap{"message": fmt.Sprintf("%d alert definitions paused", cmd.ResultCount)})
 }
 
 // alertDefinitionUnpauseEndpoint handles POST /api/alert-definitions/unpause.
-func (api *apiImpl) alertDefinitionUnpauseEndpoint(c *models.ReqContext, cmd updateAlertDefinitionPausedCommand) response.Response {
+func (api *API) alertDefinitionUnpauseEndpoint(c *models.ReqContext, cmd ngmodels.UpdateAlertDefinitionPausedCommand) response.Response {
 	cmd.OrgID = c.SignedInUser.OrgId
 	cmd.Paused = false
 
-	err := api.store.updateAlertDefinitionPaused(&cmd)
+	err := api.Store.UpdateAlertDefinitionPaused(&cmd)
 	if err != nil {
 		return response.Error(500, "Failed to unpause alert definition", err)
 	}
 	return response.JSON(200, util.DynMap{"message": fmt.Sprintf("%d alert definitions unpaused", cmd.ResultCount)})
 }
+
+// LoadAlertCondition returns a Condition object for the given alertDefinitionID.
+func (api *API) LoadAlertCondition(alertDefinitionUID string, orgID int64) (*eval.Condition, error) {
+	q := ngmodels.GetAlertDefinitionByUIDQuery{UID: alertDefinitionUID, OrgID: orgID}
+	if err := api.Store.GetAlertDefinitionByUID(&q); err != nil {
+		return nil, err
+	}
+	alertDefinition := q.Result
+
+	err := api.Store.ValidateAlertDefinition(alertDefinition, true)
+	if err != nil {
+		return nil, err
+	}
+
+	return &eval.Condition{
```

Comment:
Doesn't have to be in this PR, but I'm wondering if it would be much trouble to change `eval.Condition`s properties to be "Condition" and "Data" so it lines up with the API. (Could add property comments on the struct since the names are perhaps less descriptive, but I feel like consistency is better. thoughts?

Reply:
It shouldn't be much trouble since the modifications will be mostly in the `eval` package; therefore I would suggest merging this now and address this in another PR.
BTW since your review, I've moved `AlertQuery` from the `eval` package to `models` in case you want to have a fresh look.